ddtrace/appsec/processor.py

```diff
@@ -100,6 +100,17 @@ def _set_headers(span, headers):
             span.set_tag(_normalize_tag_name("request", k), headers[k])
 
 
+def _set_cookies(span, cookies):
+    # type: (Span, Dict[str, Union[str, List[str]]]) -> None
+    for k in cookies:
+        cookie = cookies[k]
+        # flask~=0.12.0 create a list of values for each cookie instead a string
+        if (isinstance(cookie, list) or isinstance(cookie, tuple)) and len(cookie) == 1:
+            cookie = cookie[0]
```

Comment:
can cookies be supplied multiple times? would there ever be more than 1 here?

Reply:
Flask store one cookie:
```
curl -v --cookie "USER_ATTACK=abc;USER_ATTACK=dfg" http://localhost:5000/

@app.route('/')
def check_cookie():
    print(request.cookies["USER_ATTACK"])
>> 'abc'
```

Flask stores this data in InmutableMultiDict but this class changes the behavior between versions:

Flask 0.12:
```
dict(request.cookies)
>> {'USER_ATTACK': [u"1' or '1' = '1'", u'blablabla']}
dict(request.cookies["USER_ATTACK"])
>> 'abc'
```

Flask >0.12:
```
dict(request.cookies)
>> {'USER_ATTACK': "1' or '1' = '1'"}
dict(request.cookies["USER_ATTACK"])
>> 'abc'
```